packages/coil-extension/src/popup/context/tipContext.tsx

```diff
@@ -1,40 +1,100 @@
 import React, { createContext, useContext, useState, useMemo } from 'react'
 import { StorageService } from '@webmonetization/wext/services'
 
+import { getCreditCardFromPaymentMethods } from '../../util/getCreditCardFromPaymentMethods'
+import { calculateMaxAllowableTip } from '../../util/calculateMaxAllowableTip'
+
 //
 // Models
 //
 interface ITipContext {
   currentTipAmount: number
+  maxAllowableTipAmount: number //* the maxAllowableTip is primarily responsible for disabling tipping inputs
   setCurrentTipAmount: (amount: number) => void
 }
+
 interface ITipProvider {
   storage: Pick<StorageService, 'get'>
 }
 
+//
```

Comment:
This is duplicate code of what's in the utils directory.

Reply:
Yeah i was in the process of adding the tests and moving the calc to a util when i had to fix something for the demo